components/TextControls/README.md

```diff
@@ -0,0 +1,170 @@
+  - [Installation with CocoaPods](#installation-with-cocoapods)
+  - [Importing](#importing)
+- [Usage](#usage)
+  - [TextFields](#textfields)
+- [Theming](#theming)
+  - [Theming](#theming)
+- [Examples](#examples)
+  - [Creating a textfield](#creating-a-textfield)
+
+- - -
+
+## Overview
+
+At this time, the only text control we offer is the textfield. There are three textfield classes:
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--link">MDCFilledTextField: A textfield implementing the Material <a href="https://material.io/components/text-fields/#filled-text-field">filled style</a></li>
+  <li class="icon-list-item icon-list-item--link">MDCOutlinedTextField: A textfield implementing the Material <a href="https://material.io/components/text-fields/#outlined-text-field">outlined style</a></li>
+  <li class="icon-list-item icon-list-item">MDCBaseTextField: An unstyled textfield that the previous two inherit from</li>
+</ul>
+
+## Installation
+
+<!-- Extracted from docs/installation.md -->
+
+### Installation with CocoaPods
+
+Add the following to your `Podfile`:
+
+```bash
+pod 'MaterialComponents/TextControls'
+```
+<!--{: .code-renderer.code-renderer--install }-->
+
+Then, run the following command:
+
+```bash
+pod install
+```
+
+### Importing
+
+To use TextControls in your code, import the MaterialTextControls umbrella header (Objective-C) or MaterialComponents module (Swift).
+
+<!--<div class="material-code-render" markdown="1">-->
+#### Swift
+
+```swift
+import MaterialComponents.MaterialTextControls
+```
+
+#### Objective-C
+
+```objc
+#import "MaterialTextControls.h"
+```
+
+<!--</div>-->
+
+
+
+## Usage
+
+<!-- Extracted from docs/typical-use.md -->
+
+### TextFields
+
+The largest difference between MDCTextControl textfields and UITextFields from a usability standpoint relates to the sizing behavior of MDCTextControl textfields, and the implications of that sizing behavior for layout. Where UITextField can be whatever height a user wants it to be, MDCTextControl textfields have heights that they need to be in order to look correct. The process for ensuring that MDCTextControl textfields have their preferred heights depends on whether one is in an Auto Layout or Manual Layout environment. In an Auto Layout environment, the textfield's preferred height will be reflected in `intrinsicContentSize`, and the user will not have to do anything other than set a width constraint on the textfield to ensure that the preferred height is achieved. In a Manual Layout environment, standard methods like `sizeThatFits:` or `sizeToFit` must be used to inform the frames of the textfield. These methods assume that the textfield already has the preferred width.
+
+
+## Theming
+
+<!-- Extracted from docs/theming.md -->
+
+### Theming
+
+You can theme a textfield to match the Material Design style by using a theming extension. The content below assumes you have read the article on [Theming](../../docs/theming.md).
```

Comment:
Minor copy-editing suggestion throughout here: consider referring to "textfield" either as "text field" (to refer to it as a general concept) or TextField (to refer to it as a specific component).

Reply:
good catch!